rplugin/python3/deoplete/filter/matcher_fuzzy.py

```diff
@@ -18,24 +16,18 @@ def __init__(self, vim):
         self.name = 'matcher_fuzzy'
         self.description = 'fuzzy matcher'
 
-    def filter(self, context):
+    def filter(self, context, ensure_same_prefix=True):
         complete_str = context['complete_str']
         if context['ignorecase']:
             complete_str = complete_str.lower()
 
-        if context['is_sorted']:
-            begin = binary_search_begin(
-                context['candidates'], complete_str[0])
-            end = binary_search_end(
-                context['candidates'], complete_str[0])
-            if begin < 0 or end < 0:
-                return []
-            candidates = context['candidates'][begin:end+1]
-        else:
-            candidates = context['candidates']
```

Comment:
Why you have removed the binary search feature?
Please describe it.

Reply:
Hmmm...I'm not sure what is the binary search for. I thought it's for performance purpose, isn't it?

And I thought there is a bug of the binary search in the current fuzzy matcher. It doesn't reflect to the `ignorecase` option correctly.
Thus, the test case below will fail:
```python
    ctx = {
        'complete_str': 'fobr',
        'ignorecase': True,
        'camelcase': False,
        'is_sorted': False,
        'candidates': [
            { 'word': 'FooBar' },
            { 'word': 'foobar' },
            { 'word': 'aFooBar' },
            { 'word': 'fooBar' },
        ]
    }
    assert f.filter(ctx) == [
        { 'word': 'FooBar' },
        { 'word': 'foobar' },
        { 'word': 'fooBar' },
    ]
```
I will provide profiling results later, hmm.... maybe tomorrow.